toolkit/tools/imagecustomizerapi/imageFormat.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerapi
+
+import (
+	"fmt"
+	"slices"
+	"strings"
+)
+
+type ImageFormat string
+
+const (
+	ImageFormatVhd      ImageFormat = "vhd"
+	ImageFormatVhdFixed ImageFormat = "vhd-fixed"
+	ImageFormatVhdx     ImageFormat = "vhdx"
+	ImageFormatQcow2    ImageFormat = "qcow2"
+	ImageFormatRaw      ImageFormat = "raw"
+	ImageFormatIso      ImageFormat = "iso"
+	ImageFormatCosi     ImageFormat = "cosi"
+)
+
+// supportedImageFormats is a list of all image formats defined above.
+var supportedImageFormats = []string{
+	string(ImageFormatVhd),
+	string(ImageFormatVhdFixed),
+	string(ImageFormatVhdx),
+	string(ImageFormatQcow2),
+	string(ImageFormatRaw),
+	string(ImageFormatIso),
+	string(ImageFormatCosi),
+}
+
+func (f *ImageFormat) UnmarshalYAML(unmarshal func(interface{}) error) error {
```

Comment:
Please follow the existing patterns for string enum types, like `FileSystemType`. Keeping code consistent makes it easier to maintain.

Also, having the `IsValid` function be separate from the unmarshaling code was an intentional design choice. The imager tool handled verification and unmarshalling together and I found it made the code pretty annoying.

Reply:
The enum types are mostly an internal implementation detail and don't show up in the API. Hence, there is less scrutiny paid to their naming.